distribution/docs/src/main/resources/_contents/_endpoints/cometd-endpoint-contents.adoc

```diff
@@ -0,0 +1,855 @@
+----
+"operations" : {
+	"download" : "http://example.com/product"
+}
+----
+
+If the value is not a URL, the client should send a message back to the server on the same topic with the operation name.
+
+Note: the ${ddf-branding} UI will interpret several values with special icons:
+* `download`
+* `retry`
+* `cancel`
+* `pause`
+* `remove`
+* `resume`
+
+|progress
+|Percentage value of activity completion
+|String
+|Integer between 0 - 100 followed by a %
+|"45%"
+
+|status
+|Enumerated value that displays the current state of the activity
+|String
+a|
+* `STARTED`,
+* `RUNNING`,
+* `FINISHED`,
+* `STOPPED`,
+* `PAUSED`, or
+* `FAILED`
+|`RUNNING`
+
+|timestamp
+|Time that the activity update was sent
+|String
+|Positive long value (seconds
+since unix epoch)
+|`1403801920875`
+
+|title
+|User-readable title for the activity update
+|String
+|Any String
+|"Download Complete"
+
+|subject
+|User who started the activity
+|String
+|Any String
+|"admin"
+
+|bytes
+|Number of bytes the activity consumed (upload or download)
+|Integer
+|Any Integer
+|11
+
+|session
+|The session ID of the user/subject
+|String
+|Any String
+|"6q2lfmmwh9dwv1c4gevboyp297"
+
+|Custom Value
+|Additional keys can be inserted by the component sending the activity notification
+|Any JSON Type
+|Any JSON
+|"{}"
+
+|===
+
+.Example: Activity update with custom 'bytes' field
+[source,json,linenums]
+----
+data: {
+ 	"category": "Product Retrieval",
+ 	"event.topics": "ddf/activities/broadcast",
+ 	"id": "a62f6666-fc41-4a19-91f1-485e73a564b5",
+ 	"message": "The requested product is being retrieved.
+		Standby.",
+ 	"operations": {
+ 		"cancel" : true
+	},
+ 	"progress": "",
+	"status": "RUNNING",
+	"timestamp": "1403801920875",
+ 	"title": "Product downloading",
+ 	"user": "admin",
+ 	"bytes": 635084800
+}
+----
+
+====== Activity Operations Channel
+
+An activity operation is published to the channel `/service/action`
+
+.CometD Activity Format
+[cols="1m,2,1,1,2,2" options="header"]
+|===
+|Map Key
+|Description
+|Value Type
+|Possible Values
+|Example Values
+|Comments
+
+|action
+|Requested action
+|String
+|Any String.
+a|Common values are
+
+* "download",
+* "retry",
+* "cancel",
+* "pause",
+* "remove",
+* "resume"
+* "cancel"
+
+|Based on the operations map that comes in from an activity event.
+
+|id
+|ID of the activity
+|String
+|Any String
+|"a62f6666-fc41-4a19-91f1-485e73a564b5"
+|The Activity ID to which the requested operation relates
+|===
+
+.Example: Activity Operation Request Message
+[source,json,linenums]
+----
+"data": [ {
+	"action":"cancel",
+ 	"id":"a62f6666-fc41-4a19-91f1-485e73a564b5"
+} ]
+----
+
+====== Query Service Channel
+
+All query requests should be published to the `/service/query` channel.
+
+When performing a CometD publish command, the data being published must be valid json with 'data' being the key to a map that contains the following values:
+
+.Query Request Format
+[cols="1m,3,1,1,1,1" options="header"]
+|===
+|Map Key
```

Comment:
Looks like `timeout` is missing from the table and `format` shouldn't be included here.

Also, we need to indicate which ones of these are required (`id`, `cql`) and which ones are optional (`src`, `timeout`, `start`, `count`).